pkg/controller/proxyconfig/proxyconfig_controller.go

```diff
@@ -0,0 +1,214 @@
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_proxyconfig")
+
+const (
+	// singletonName          = "instance"
+	// singletonMessage       = "ClusterLogging is a singleton. Only an instance named 'instance' is allowed"
+	proxyName              = "cluster"
+	trustBundleConfigmapNS = "openshift-config"
+)
+
+// Add creates a new ClusterLogging Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	if err := configv1.Install(mgr.GetScheme()); err != nil {
+		return &ReconcileProxyConfig{}
+	}
+
+	return &ReconcileProxyConfig{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("proxyconfig-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// We only care about a configmap source with a specific name/namespace,
+	// so filter events before they are provided to the controller event handlers.
+	pred := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			b := handleConfigMap(e.MetaNew)
+			logrus.Infof("DBG_PX: Update '%v'", b)
+			return b
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			b := handleConfigMap(e.Meta)
+			logrus.Infof("DBG_PX: Delete '%v'", b)
+			return b
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			b := handleConfigMap(e.Meta)
+			logrus.Infof("DBG_PX: Create '%v'", b)
+			return b
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			b := handleConfigMap(e.Meta)
+			logrus.Infof("DBG_PX: Generic '%v'", b)
+			return b
+		},
+	}
+
+	/******** DID NOT WORK *********
+	h := &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &configv1.Proxy{},
+	}
+	******** DID NOT WORK *********/
+
+	// Watch for changes to the additional trust bundle configmap.
+	if err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, pred); err != nil {
```

Comment:
this should likely be an `EnqueueRequestForOwner` since we want the proxy config enqueued not the configmap, right?

Reply:
Yes.  I'm going to replace it (it did not solve my problem, though...)

I learned how to use c.Watch for the cluster logging in cluster-network-operator, where it simply sets &handler.EnqueueRequestForObject{}.  Probably, the expected behaviour could be different from ours.
https://github.com/openshift/cluster-network-operator/blob/master/pkg/controller/proxyconfig/controller.go#L64

Reply:
Follow-up: I tested with `EnqueueRequestForOwner`, but CLO did not start with it.  So, I think we need to have `EnqueueRequestForObject` here.